git-hub

```diff
@@ -22,46 +84,40 @@ def pr(num):
     num : integer
         The number of the pull request.
     """
-    process = subprocess.Popen("git remote -v", stdout=subprocess.PIPE)
-    remotes = str(process.stdout.read())
-    url = remotes.split(" ", 1)[0]  # gets the fetch url
-    url = url.split("\\t", 1)[1]  # takes out extra info
-    arguments = url.split(".com:")[1]  # gets just the username/repo.git
-    arguments = arguments[:len(arguments) - 4]  # takes out ".git"
-    username, repo = arguments.split("/")
-
-    # gets token from config folder.
-    try:
-        with open(os.path.expanduser("~/.config/git-hub.yaml")) as stream:
-            yaml_file = str(yaml.load(stream))
-            token = yaml_file.split("=")[1].strip()
-
-    except (IndexError, FileNotFoundError) as e:
-        print(textwrap.dedent("""Needs an authentification token in: ~/.config/git-hub.yaml
-            file simply says: token = abc123
-            Visit https://help.github.com/articles/
-            creating-a-personal-access-token-for-the-command-line/"""))
-        sys.exit(1)
+    username, repo, remotes = login()
+    token = get_token()
 
     # gets pr and runs command.
     try:
         g = github.Github(token)
         pr = g.get_user(username).get_repo(repo).get_pull(num)
         label = pr.head.label
         other_user, branch = label.split(":")
+        process = subprocess.Popen(["git", "remote", "-v"], stdout=subprocess.PIPE)
+        remotes = str(process.stdout.read())
         remote_name = "\\n" + other_user + "\\t"
         if remote_name not in remotes:
-            p = subprocess.Popen(["git", "remote", "add", other_user,
-                                 f'git@github.com:{other_user}/{repo}'])
+            p = subprocess.Popen(["git", "remote", "add", other_user, f'git@github.com:{other_user}/{repo}'])
             p.communicate()
         p = subprocess.Popen(["git", "fetch", other_user],
                              stdout=subprocess.PIPE)
         p.stdout.read()
         subprocess.Popen(["git", "checkout", "-b",
                          f'pr/{num}', f'{other_user}/{branch}'])
-        with open(".git_hub_cache", "w") as outfile:
-            data = {"user": other_user, "pr": f'pr/{num}', "branch": branch}
-            json.dump(data, outfile)
+        file_path = path_to_git()
+        file_path = os.path.join(file_path, '.git_hub_cache')
+        with open(file_path, "a+") as f:
```

Comment:
There is no need to create the file here if it does not exist.  Later, you will read its contents (if the file exists), and rewrite the entire file.